js/services/addressBook_service.js

```diff
@@ -72,7 +72,111 @@ app.factory('AddressBookService', ['DavClient', 'DavService', 'SettingsService',
 				}
 				return addressBook;
 			});*/
+		},
+
+		share: function(addressBook, shareType, shareWith, writable, existingShare) {
+			var xmlDoc = document.implementation.createDocument('', '', null);
+			var oShare = xmlDoc.createElement('o:share');
+			oShare.setAttribute('xmlns:d', 'DAV:');
+			oShare.setAttribute('xmlns:o', 'http://owncloud.org/ns');
+			xmlDoc.appendChild(oShare);
+
+			var oSet = xmlDoc.createElement('o:set');
+			oShare.appendChild(oSet);
+
+			var dHref = xmlDoc.createElement('d:href');
+			if (shareType === OC.Share.SHARE_TYPE_USER) {
+				dHref.textContent = 'principal:principals/users/';
+			} else if (shareType === OC.Share.SHARE_TYPE_GROUP) {
+				dHref.textContent = 'principal:principals/groups/';
+			}
+			dHref.textContent += shareWith;
+			oSet.appendChild(dHref);
+
+			var oSummary = xmlDoc.createElement('o:summary');
+			oSummary.textContent = t('contacts', '{addressbook} shared by {owner}', {
+				addressbook: addressBook.displayName,
+				owner: addressBook.owner
+			});
+			oSet.appendChild(oSummary);
+
+			if (writable) {
+				var oRW = xmlDoc.createElement('o:read-write');
+				oSet.appendChild(oRW);
+			}
+
+			var body = oShare.outerHTML;
+
+			DavClient.xhr.send(
```

Comment:
I believe this is missing a `return`. I'm not sure if https://github.com/owncloud/contactsrework/pull/18/files#diff-dbaa024398385bf6cffe22c422471ff0R96 works without this.